packages/xstate-test/src/TestModel.ts

```diff
@@ -0,0 +1,491 @@
+import {
+  getPathFromEvents,
+  performDepthFirstTraversal,
+  SerializedEvent,
+  SerializedState,
+  SimpleBehavior,
+  StatePath,
+  StatePlan,
+  Step,
+  TraversalOptions,
+  traverseShortestPlans,
+  traverseSimplePathsTo,
+  traverseSimplePlans
+} from '@xstate/graph';
+import { EventObject, SingleOrArray } from 'xstate';
+import {
+  CoverageFunction,
+  coversAllStates,
+  coversAllTransitions
+} from './coverage';
+import { planGeneratorWithDedup } from './dedupPathPlans';
+import type {
+  CriterionResult,
+  EventExecutor,
+  GetPlansOptions,
+  PlanGenerator,
+  StatePredicate,
+  TestModelCoverage,
+  TestModelOptions,
+  TestPathResult,
+  TestStepResult
+} from './types';
+import { flatten, formatPathTestResult, simpleStringify } from './utils';
+
+export interface TestModelDefaults<TState, TEvent extends EventObject> {
+  coverage: Array<CoverageFunction<TState, TEvent>>;
+  planGenerator: PlanGenerator<TState, TEvent>;
+}
+
+export const testModelDefaults: TestModelDefaults<any, any> = {
+  coverage: [coversAllStates<any, any>(), coversAllTransitions<any, any>()],
+  planGenerator: traverseShortestPlans
+};
+
+/**
+ * Creates a test model that represents an abstract model of a
+ * system under test (SUT).
+ *
+ * The test model is used to generate test plans, which are used to
+ * verify that states in the `machine` are reachable in the SUT.
+ *
+ */
+export class TestModel<TState, TEvent extends EventObject> {
+  private _coverage: TestModelCoverage<TState, TEvent> = {
+    states: {},
+    transitions: {}
+  };
+  public options: TestModelOptions<TState, TEvent>;
+  public defaultTraversalOptions?: TraversalOptions<TState, TEvent>;
+  public getDefaultOptions(): TestModelOptions<TState, TEvent> {
+    return {
+      serializeState: (state) => simpleStringify(state) as SerializedState,
+      serializeEvent: (event) => simpleStringify(event) as SerializedEvent,
+      getEvents: () => [],
+      states: {},
+      events: {},
```

Comment:
if you like the proposal to rename `states` to `stateAssertions` then this could be renamed too but I don't have any particular proposal for a better name myself

Reply:
See above comment